go-controller/pkg/node/secondary_node_network_controller.go

```diff
@@ -122,11 +122,37 @@ func (oc *SecondaryNodeNetworkController) getNetworkID() (int, error) {
 	return *oc.networkID, nil
 }
 
+func (oc *SecondaryNodeNetworkController) shouldReconcileNetworkChange(old, new util.NetInfo) bool {
+	wasUDNNetworkAdvertisedAtNode := util.IsPodNetworkAdvertisedAtNode(old, oc.name)
+	isUDNNetworkAdvertisedAtNode := util.IsPodNetworkAdvertisedAtNode(new, oc.name)
+	return wasUDNNetworkAdvertisedAtNode != isUDNNetworkAdvertisedAtNode
+}
+
+// Reconcile function reconciles three entities based on whether UDN network is advertised
+// and the gateway mode:
+// 1. IP rules
+// 2. OpenFlows on br-ex bridge to forward traffic to correct ofports
 func (oc *SecondaryNodeNetworkController) Reconcile(netInfo util.NetInfo) error {
-	// reconcile network information, point of no return
+	reconcilePodNetwork := oc.shouldReconcileNetworkChange(oc.ReconcilableNetInfo, netInfo)
+
 	err := util.ReconcileNetInfo(oc.ReconcilableNetInfo, netInfo)
 	if err != nil {
 		klog.Errorf("Failed to reconcile network %s: %v", oc.GetNetworkName(), err)
 	}
+
+	if reconcilePodNetwork {
+		isUDNNetworkAdvertisedAtNode := oc.isUDNNetworkAdvertisedAtNode()
+		if oc.gateway != nil {
+			oc.gateway.SetUDNNetworkAdvertised(oc.GetNetworkName(), isUDNNetworkAdvertisedAtNode)
+			if err := oc.gateway.Reconcile(); err != nil {
+				return err
```

Comment:
Technically we should just klog an error here, since ReconcileNetInfo above is a point of no return (I had a comment on the original code, perhaps you can add it back) as once it is called. NeedsReconciliation will return false anyway.

Reply:
Yes, below the point of no return which is `ReconcileNetInfo` call.